arc/database/api.py

```
@@ -741,6 +759,121 @@
+                """), batch_data)
+            
+            logger.info(f"Recorded {len(tool_usage_records)} tool usage records in batch")
+            return usage_ids
+            
+        except Exception as e:
+            logger.error(f"Failed to record tool usage batch: {e}")
+            raise DatabaseError(f"Failed to record tool usage batch: {e}") from e
+    
+    async def record_batch_failures(
+        self,
+        failure_records: List[Dict[str, Any]]
```

Ruff (0.11.9)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)